Update IOU Preview Card Styles

grgia opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
Updating the IOU Preview Styles 
![image](https://user-images.githubusercontent.com/2319350/202061924-d6fb51c8-546b-4c43-8a49-02489e53716a.png)

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/12258
$ https://github.com/Expensify/App/issues/12958



### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
<img width="1223" alt="image" src="https://user-images.githubusercontent.com/38015950/202827970-b3d6ebd2-a5f0-4121-94ba-34da6ef77572.png">
Set to offline, card looks good + highlight still works

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
- [ ] Verify that no errors appear in the JS console

To test in Dark Mode. Change the export line in `default.js` to `export default darkTheme;`

1. Have account A request money from account B
2. Cancel a request from account A
3. Have account A request money from account B
4. Decline a request from account B
5. Have account A initiate a Split with Account B and account C
6. Pay a request from account A from account B

- [ ] Ensure styles are correct (see Shawn's mockups above)
- [ ] ensure color changes when highlighted
- [ ] ensure clicking on preview box opens modal, ensure clicking on arrow opens modal

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).

- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for expected offline behavior are in the `Offline steps` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [ ] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->
| Light | Light Hover | Dark | Dark Hover |
| ---- | ---- | ---- | ---- |
| <img width="1223" alt="image" src="https://user-images.githubusercontent.com/38015950/202825803-04528840-701f-43ae-8b5a-4b9ea013a8a6.png">|<img width="1223" alt="image" src="https://user-images.githubusercontent.com/38015950/202825905-dcab3ee3-1f0a-45d7-b233-9fa935f1fdcd.png">|<img width="1223" alt="image" src="https://user-images.githubusercontent.com/38015950/202825839-9bf67d92-25c3-4690-9653-21340494153f.png">|<img width="1223" alt="image" src="https://user-images.githubusercontent.com/38015950/202825849-910c1bac-d69d-40f7-9617-52cbe434785b.png">|
#### Mobile Web - Chrome
<!-- Insert screenshots of your changes on the web platform (from chrome mobile browser)-->
| Light | Dark |
| ---- | ---- |
| <img width="502" alt="image" src="https://user-images.githubusercontent.com/38015950/202826629-f24fd22f-8f61-49e1-b7a3-f3985322f829.png">|<img width="502" alt="image" src="https://user-images.githubusercontent.com/38015950/202826790-600548fe-7d5e-4aa3-a67a-ee9c73b85e12.png">| 
#### Mobile Web - Safari
<!-- Insert screenshots of your changes on the web platform (from safari mobile browser)-->
| Light | Dark |
| ---- | ---- |
|<img width="620" alt="image" src="https://user-images.githubusercontent.com/38015950/202827051-b0435db2-2f7d-44cd-89ea-1873d1645b0c.png">|<img width="620" alt="image" src="https://user-images.githubusercontent.com/38015950/202827024-016121e2-90fa-4498-972c-21f64ff4b08f.png">| 
#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->
| Light | Light Hover | Dark | Dark Hover |
| ---- | ---- | ---- | ---- |
|<img width="1312" alt="image" src="https://user-images.githubusercontent.com/38015950/202826068-79a90baa-f252-47c4-8bc6-bac4bb6f0cb1.png">|<img width="1312" alt="image" src="https://user-images.githubusercontent.com/38015950/202826078-8734ab0c-63b8-4e48-9fc7-8e4cc75fc960.png">|<img width="1312" alt="image" src="https://user-images.githubusercontent.com/38015950/202826146-f5163dfb-bd41-4c2f-a779-d70df8af7391.png">|<img width="1312" alt="image" src="https://user-images.githubusercontent.com/38015950/202826164-89095ff1-fa25-4846-8028-827ef77cb8e1.png">|
#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->
| Light | Dark |
| ---- | ---- |
|<img width="620" alt="image" src="https://user-images.githubusercontent.com/38015950/202827079-a02e22a6-ecfc-478e-99bb-8d47f870385c.png">|<img width="620" alt="image" src="https://user-images.githubusercontent.com/38015950/202827000-839b0a13-895c-452a-bb7f-6c1f6491d48f.png">| 
#### Android
<!-- Insert screenshots of your changes on the Android platform-->
| Light | Dark |
| ---- | ---- |
|<img width="502" alt="image" src="https://user-images.githubusercontent.com/38015950/202826653-19a60f83-d73f-4b20-ab74-ed33234f0749.png">|<img width="502" alt="image" src="https://user-images.githubusercontent.com/38015950/202826758-8525c009-0196-4e79-80c5-e4bd8c3175eb.png">|

grgia committed:
Remove "View Details" from IOU and add ">" icon

grgia committed:
Merge branch 'main' into georgia-IOU-branding, fix styles

grgia committed:
Remove unused

grgia commented on src/styles/styles.js:
```diff
@@ -2165,9 +2165,7 @@ const styles = {
-        backgroundColor: themeColors.componentBG,
-        borderColor: themeColors.border,
-        borderWidth: 1,
+        backgroundColor: themeColors.modalBackground,
```
will update with new colors

grgia committed:
Combine Imports, Align Start

shawnborton commented:
Looks like that gray is just one shade too dark (should be gray1, FWIW). That being said, maybe we should handle the row hover case and make the BG darker on hover? This is similar to what we would do for the workspace names in pills in the LHN on hover. 

Also, looks like we need to handle the border outline for the avatars too.

grgia committed:
Merge branch 'main' into georgia-IOU-branding

grgia commented:
<img width="1170" alt="image" src="https://user-images.githubusercontent.com/38015950/202798860-d538ce61-5a5c-449b-bcb8-0b6fbbfde076.png">
Here's what it looks like with gray1, working on your other suggestions now

shawnborton commented:
Nice, that looks good!

grgia committed:
Fix colors of card, avatar border

Luke9389 commented on src/components/ReportActionItem/IOUQuote.js:
```diff
@@ -1,46 +1,48 @@
                         onPress={props.onViewDetailsPressed}
                     >
-                        {props.translate('iou.viewDetails')}
+                        {Str.htmlDecode(fragment.text.split(' ')[0])}
```
When I first saw this, I thought this would cause breaking behavior with translations (i.e. not translating). But then I tested it, and it totally works! So nice catch! Do you happen to know why this works? I'm thinking the message must get translated upstream, but I went as far as ReportActionItem and didn't see the call to `translate` anywhere. 

Anyway, now that we don't need this method here, we can also remove the import for withLocalize, the withLocalize props, and just export the component with `export default IOUQuote` instead of passing it to withLocalize. 

Nice catch!

grgia replied:
https://github.com/Expensify/App/blob/main/src/libs/ReportUtils.js#L703-L729
This is where the fragment is coming from! Which brings up a great point- this isn't using translate. I think I can fix that  here as well?

grgia replied:
<img width="1635" alt="image" src="https://user-images.githubusercontent.com/38015950/202818260-aa849171-5f60-4576-baec-ecf6a6c5c8ed.png">
In spanish ^^

grgia replied:
Commenting this for reference later, I think since these are considered a report action, it doesn't update with localize? I can get it to work on send, but not on page load, refresh, or switching languages- it always reverts to english. For now, I'm going to ignore since the method you pointed out would still work if we switched languages

in en.js iou:
``` js
        requestedAction: ({amount, comment, who}) => `Requested ${amount} from ${who}${comment && ` for ${comment}`}`,
        splitAction: ({amount, comment, who}) => `Split ${amount} with ${who}${comment && ` for ${comment}`}`,
        cancelledAction: ({amount, comment}) => `Cancelled the ${amount} request${comment && ` for ${comment}`}`,
        declinedAction: ({amount, comment}) => `Declined the ${amount} request${comment && ` for ${comment}`}`,
        paidAction: ({amount, comment, who}) => `Paid ${amount} to ${who}${comment && ` for ${comment}`}`,
```

in es.js iou:
``` js
        requestedAction: ({amount, comment, who}) => `Solicitó ${amount} de ${who}${comment && ` por ${comment}`}`,
        splitAction: ({amount, comment, who}) => `Split ${amount} con ${who}${comment && ` por ${comment}`}`,
        cancelledAction: ({amount, comment}) => `Canceló la solicitud de ${amount} ${comment && ` por ${comment}`}`,
        declinedAction: ({amount, comment}) => `Se rechazó la solicitud de ${amount} ${comment && ` por ${comment}`}`,
        paidAction: ({amount, comment, who}) => `Pagué ${amount} a ${who}${comment && ` por ${comment}`}`,
```

in reportUtils getIOUReportActionMessage()
``` js
    let iouMessage;
    switch (type) {
        case CONST.IOU.REPORT_ACTION_TYPE.CREATE:
            iouMessage = Localize.translateLocal('iou.requestedAction', {amount, who, comment});
            break;
        case CONST.IOU.REPORT_ACTION_TYPE.SPLIT:
            iouMessage = Localize.translateLocal('iou.splitAction', {amount, who});
            break;q
        case CONST.IOU.REPORT_ACTION_TYPE.CANCEL:
            iouMessage = Localize.translateLocal('iou.cancelledAction', {amount, who, comment});
            break;
        case CONST.IOU.REPORT_ACTION_TYPE.DECLINE:
            iouMessage = Localize.translateLocal('iou.declinedAction', {amount, who});
            break;
        case CONST.IOU.REPORT_ACTION_TYPE.PAY:
            iouMessage = Localize.translateLocal('iou.paidAction', {amount, who, comment});
            break;
        default:
            break;
    }
```

Luke9389 reviewed (dismissed):
I love how the redesign seems to be removing styles (most of the time). PR looks good except for the un-needed imports 👍

grgia committed:
Make hover change color of preview box and multiple avatars

grgia commented:
**Hover:**
<img width="1296" alt="image" src="https://user-images.githubusercontent.com/38015950/202824386-09e4486d-4dc3-4f92-88a0-09c54219f8d7.png">
<img width="1296" alt="image" src="https://user-images.githubusercontent.com/38015950/202824402-8450bcfe-7e6c-4b8e-a8d7-d8cdde8c2006.png">

<img width="1296" alt="image" src="https://user-images.githubusercontent.com/38015950/202824431-7ac17ae1-43c6-40b6-8010-15575eb6040d.png">
<img width="1296" alt="image" src="https://user-images.githubusercontent.com/38015950/202824444-30366196-fb2d-47e3-babe-0bf0cf20a584.png">

shawnborton commented:
Nice, I think that will work great!

grgia commented:
@shawnborton taking this one to the finish line!

parasharrajat commented on src/pages/home/report/ReportActionItem.js:
```diff
@@ -161,6 +127,44 @@ class ReportActionItem extends Component {
+        this.setState({isContextMenuActive: ReportActionContextMenu.isActiveReportAction(this.props.action.reportActionID)});
+    }
+
+    /**
+                 * Show the ReportActionContextMenu modal popover.
+                 *
+                 * @param {Object} [event] - A press event.
+                 */
```
let's fix this.

parasharrajat commented on src/pages/home/report/ReportActionItem.js:
```diff
@@ -97,49 +97,15 @@ class ReportActionItem extends Component {
-            return <RenameAction action={this.props.action} />;
-        }
-
+    getChildren(hovered = false) {
```
```suggestion
    renderItemContent(hovered = false) {
```

Please add jsDoc

parasharrajat commented on src/styles/styles.js:
```diff
@@ -2171,16 +2171,18 @@ const styles = {
     },
 
+    iouPreviewBoxHover: {
+        backgroundColor: themeColors.border,
```
This is a wrong variable. Either create a new one or use some other variable. Using wrong theme variable makes it harder to customize the theme.

grgia replied:
Good point, I found a better descriptive color. Just a heads up, we don't want to add any new theme variables until we switch to the new color mode

parasharrajat replied:
Got it.

parasharrajat commented on src/components/ReportActionItem/IOUQuote.js:
```diff
@@ -1,46 +1,48 @@
+                    <Text style={[styles.chatItemMessage]}>
+                        {Str.htmlDecode(fragment.text.substring(fragment.text.indexOf(' ')))}
+                    </Text>
+                </Text>
```
```suggestion
```

parasharrajat commented on src/components/ReportActionItem/IOUQuote.js:
```diff
@@ -1,46 +1,48 @@
-                </Text>
-                {props.shouldShowViewDetailsLink && (
+            <View key={`iouQuote-${props.action.reportActionID}-${index}`} style={[styles.textInputAndIconContainer, styles.alignItemsStart, styles.justifyContentBetween]}>
+                <Text>
```
```suggestion
```

parasharrajat commented on src/components/ReportActionItem/IOUPreview.js:
```diff
@@ -159,7 +162,10 @@ const IOUPreview = (props) => {
                                         <MultipleAvatars
                                             icons={[managerAvatar, ownerAvatar]}
-                                            secondAvatarStyle={[styles.secondAvatarInline]}
+                                            secondAvatarStyle={[styles.secondAvatarInline, {borderColor: themeColors.cardBG},
+                                                props.isFocused
+                                                    ? {borderColor: themeColors.border}
+                                                    : undefined]}
```
No inline styles. We can move these to styles.js 

A clean syntax for conditional styles will be 
```
[
	props.isFocused && {borderColor: themeColors.border}
]
```

parasharrajat commented on src/components/ReportActionItem/IOUPreview.js:
```diff
@@ -65,6 +65,8 @@ const propTypes = {
         hasOutstandingIOU: PropTypes.bool,
     }),
 
+    isFocused: PropTypes.bool,
```
Missing comment.

parasharrajat commented on src/components/ReportActionItem/IOUAction.js:
```diff
@@ -53,7 +57,11 @@ const IOUAction = (props) => {
                         onPayButtonPressed={launchDetailsModal}
                         onPreviewPressed={launchDetailsModal}
-                        containerStyles={[styles.cursorPointer]}
+                        containerStyles={[styles.cursorPointer,
+                            props.isFocused
+                                ? styles.iouPreviewBoxHover
+                                : undefined]}
```
```suggestion
                        containerStyles={[
                        	styles.cursorPointer,
                            	props.isFocused && styles.iouPreviewBoxHover
                            ]}
```

grgia replied:
I had to keep the original ternary (it's used in a few other files) because I kept getting console errors as react wouldn't accept the false boolean as a style, even if in an array or object

parasharrajat commented:
What is the interactive element here?
a) The whole message. 
b) Arrow. 

Previously, it was ` view details` link and the card.


I think it should be full message now and the card. Only Arrow does not fit with our other components. Take menu items for example. They have right arrow but the whole elment is clickable.

grgia committed:
Remove inline styling, use more descriptive theme colors, fix comments

grgia commented:
I addressed most of these suggestions, and I agree that we should follow interactive patterns with the arrow. I'll wait for @shawnborton to confirm before I fix the IOUQuote file. (arrow, message, or entire row)

parasharrajat commented on src/pages/home/report/ReportActionItem.js:
```diff
@@ -121,25 +125,20 @@ class ReportActionItem extends Component {
-
+    /**
+     * Get the content of ReportActionItem
+     * @param {Bool} hovered whether the ReportActionItem is hovered
```
```suggestion
     * @param {Boolean} hovered whether the ReportActionItem is hovered
```

parasharrajat commented:
Bump @shawnborton https://github.com/Expensify/App/pull/12762#issuecomment-1320766310

shawnborton commented:
Yeah, I think it's fine if that whole message is interactive and clicking anywhere on that row takes you to the IOU view.

parasharrajat reviewed (changes requested):
BUG: Tick mark is misplaced. 
![Screenshot 2022-11-21 22:59:42](https://user-images.githubusercontent.com/24370807/203121724-292d126b-37b5-45c6-8ece-e6443c354e8e.png)

parasharrajat commented:
Regarding, https://github.com/Expensify/App/pull/12762#issuecomment-1320766310. 

I noticed that we also have links like `sent` `requested` which open the page. But Having both links and full-row interactivity is looking better.

parasharrajat commented:
Are we updating the Pay button in this PR? 
![image](https://user-images.githubusercontent.com/24370807/203123165-5fe4a53a-dca6-4c67-9742-4e25ffd5eefe.png)

Current on this PR. 

![Screenshot 2022-11-21 23:08:18](https://user-images.githubusercontent.com/24370807/203123299-12e915f1-6275-4cb7-8b6d-bbefc5f39d71.png)

shawnborton commented:
Button shapes will be updated in a different PR (cc @Luke9389)

parasharrajat commented:
How can I switch to the Dark theme? Please add that step for the QA as well. Should we add a test preference for QA?
![Screenshot 2022-11-21 23:12:02](https://user-images.githubusercontent.com/24370807/203124107-c64f9a46-58d3-479d-bcce-9a82e92b687b.png)

parasharrajat commented on src/styles/styles.js:
```diff
@@ -2194,6 +2196,10 @@ const styles = {
     },
 
+    iouPreviewBoxAvatarHover: {
+        borderColor: themeColors.activeComponentBG,
```
**Not a blocker**: I think we should use a border variable here for easy customizability.

grgia commented:
@parasharrajat I added QA steps on switching to dark theme! Looking at your suggestions now

grgia commented:
@shawnborton I just want to clarify-
Currently to open the IOU Preview Modal you would click on "View Details" or on the card.
When you say entire row, do you mean clicking outside the card/message would bring you to the modal? (i.e the blue "Requested" word isn't technically a link) Would we want the user to be able to click anywhere on that row, even outside the message/card?

grgia commented on src/components/ReportActionItem/IOUAction.js:
```diff
@@ -26,12 +26,17 @@ const propTypes = {
         /** The participants of this report */
         participants: PropTypes.arrayOf(PropTypes.string),
     }),
+
+    /** Whether this option is currently in focus so we can modify its style */
+    isFocused: PropTypes.bool,
+
```
good call!

grgia committed:
Change isFocused to isHovered

shawnborton commented:
That's correct - I think it would be nice if you could click anywhere on that row, especially since it has an arrow off to the right.

parasharrajat commented:
My thought is that we exclude the card's outer area from clicking. So only the area marked as red will open the details page. 

![Screenshot 2022-11-22 at 00-43-02 Update IOU Preview Card Styles by grgia · Pull Request #12762 · Expensify_App](https://user-images.githubusercontent.com/24370807/203140063-80eb1f73-2cce-4b94-9799-08a6e4c09637.png)

grgia committed:
Merge branch 'main' into georgia-IOU-branding

grgia committed:
Fix check

grgia committed:
Make Message and IOU card clickable

grgia committed:
Style cannot be boolean, fixing console error.

grgia committed:
Update src/pages/home/report/ReportActionItem.js

Co-authored-by: Rajat Parashar <parasharrajat@users.noreply.github.com>

grgia commented:
<img width="556" alt="image" src="https://user-images.githubusercontent.com/38015950/203178016-056a2b8d-2bb1-418a-a30f-01a52170e2ad.png">
<img width="930" alt="image" src="https://user-images.githubusercontent.com/38015950/203178106-f31d70f3-8212-443b-a989-edf7a2c965cc.png">

<img width="566" alt="image" src="https://user-images.githubusercontent.com/38015950/203178240-ba21d1cd-ce21-4108-8904-19dbd1b03955.png">
<img width="897" alt="image" src="https://user-images.githubusercontent.com/38015950/203178482-ac458851-492b-43a5-8ee3-e699854b64f1.png">

shawnborton commented:
Looks great!

parasharrajat reviewed (changes requested):
BUG: There should be no Arrow and link for the split messages. Also, the message should not be clickable for split bills. 

Steps: 
1. Create a group between two users. 
2. Now Open that group report. 
3. Click :heavy_plus_sign: . 
4. Click Split Bill and split the bill. 
5. Now check the history of messages. 
6. All Split messages should not be clickable. 

![image](https://user-images.githubusercontent.com/24370807/203289813-60c5765a-2772-4185-8863-485881e1cdb1.png)


You can run the tests from https://github.com/Expensify/App/pull/2394 to confirm.

parasharrajat commented on src/components/ReportActionItem/IOUAction.js:
```diff
@@ -42,7 +47,6 @@ const IOUAction = (props) => {
         <>
             <IOUQuote
                 action={props.action}
-                shouldShowViewDetailsLink={Boolean(props.action.originalMessage.IOUReportID)}
```
BUG https://github.com/Expensify/App/pull/12762#pullrequestreview-1189779036 is happening due to the removal of this.  

Maybe rename this to `shouldAllowViewDetails`.

grgia replied:
``` js
const IOUQuote = props => (
    <View style={[styles.chatItemMessage]}>
        {_.map(props.action.message, (fragment, index) => (
            <Pressable
                key={`iouQuote-${props.action.reportActionID}-${index}`}
                onPress={props.shouldAllowViewDetails
                    ? props.onViewDetailsPressed
                    : () => {}}
                style={[styles.flexRow, styles.justifyContentBetween]}
            >
                <Text>
                    <Text style={[styles.chatItemMessageLink]}>
                        {/* Get first word of IOU message */}
                        {Str.htmlDecode(fragment.text.split(' ')[0])}
                    </Text>
                    <Text style={[styles.chatItemMessage]}>
                        {/* Get remainder of IOU message */}
                        {Str.htmlDecode(fragment.text.substring(fragment.text.indexOf(' ')))}
                    </Text>
                </Text>
                <Icon src={Expensicons.ArrowRight} fill={props.shouldAllowViewDetails ? themeColors.icon : themeColors.transparent} />
            </Pressable>
        ))}
    </View>
);
```

Do you have input on the best method of doing this? Three things change when a split bill is not clickable: (1) the icon is removed/transparent, (2) the cursor should no longer be a pointer, and (3) clicking should not open the modal. 
This solution works for (1) and (3) but the cursor is still in focus mode because of the pressable parent component. Should I use a function here instead of 3 ternaries?

parasharrajat replied:
(1) the icon is removed/transparent, :heavy_check_mark: 
 (2) the cursor should no longer be a pointer :heavy_check_mark: 
 (3) clicking should not open the modal. :heavy_check_mark: 
 
 All should be done. 
 
 For 2, 
 1. We can't use disabled prop, it disables selection. 
 2. Instead, we can use `focusable={false}` and apply a new style `{cursor: 'default'}`.

parasharrajat commented on src/components/ReportActionItem/IOUQuote.js:
```diff
@@ -1,46 +1,49 @@
-                        onPress={props.onViewDetailsPressed}
-                    >
-                        {props.translate('iou.viewDetails')}
+            <View
+                key={`iouQuote-${props.action.reportActionID}-${index}`}
+            >
+                <Pressable onPress={props.onViewDetailsPressed} style={[styles.flexRow, styles.justifyContentBetween]}>
```
We can merge these together.

grgia committed:
Make split bill not pressable

grgia committed:
Make split not clickable, fix cursor style.

grgia committed:
Merge branch 'georgia-IOU-branding' of github.com:Expensify/App into georgia-IOU-branding

parasharrajat commented on src/components/ReportActionItem/IOUQuote.js:
```diff
@@ -15,36 +15,47 @@ const propTypes = {
     /** All the data of the action */
     action: PropTypes.shape(reportActionPropTypes).isRequired,
 
+    /** Should the View Details link be displayed? */
```
```suggestion
    /** Whether it is allowed to view details. */
```

grgia committed:
Remove unrelated change from another branch

parasharrajat commented:
BUG: WEB: The highlighted background color is not removed on hovering out.

Steps: 
1. Open a report on the web. 
2. Right-click a message to open the context menu. 
3. Close the menu by clicking anywhere on the app. 
4. Notice that the message is still highlighted even if the mouse is out. 


https://user-images.githubusercontent.com/24370807/203413446-6199d1a9-8224-4571-80b5-f8c5a5d3291d.mp4

parasharrajat commented on src/pages/home/report/ReportActionItem.js:
```diff
@@ -121,25 +125,20 @@ class ReportActionItem extends Component {
+     * @param {Boolean} hovered whether the ReportActionItem is hovered
+     * @returns {Object} child component(s)
+     */
+    renderItemContent(hovered = false) {
```
Did we bind this method? I think it needs to be bound to the component instance.

grgia replied:
Good catch, I think that fixed the hover bug!

grgia committed:
Bind renderItemContent to ReportActionItem component

grgia committed:
Update src/components/ReportActionItem/IOUQuote.js

Co-authored-by: Rajat Parashar <parasharrajat@users.noreply.github.com>

parasharrajat commented:
Testing...

parasharrajat commented:
I checked [this issue](https://github.com/Expensify/App/pull/12762?notification_referrer_id=NT_kwDOAXPed7M0ODQ1NjEwNzc5OjI0MzcwODA3#issuecomment-1324196015) exists on the main so we are good here.

--- 

But there is one thing that we should fix. 

When the context menu is opened, we should keep the IOUPreview highlighted(in the hovered state). 


https://user-images.githubusercontent.com/24370807/203427954-6f21e028-5bb9-4147-ab57-76e2f4ba074a.mp4

parasharrajat commented on src/pages/home/report/ReportActionItem.js:
```diff
@@ -198,12 +208,12 @@ class ReportActionItem extends Component {
                                         ? (
                                             <ReportActionItemSingle action={this.props.action} showHeader={!this.props.draftMessage}>
-                                                {children}
+                                                {this.renderItemContent(hovered)}
```
```suggestion
                                                {this.renderItemContent(hovered || this.state.isContextMenuActive)}
```

parasharrajat commented on src/pages/home/report/ReportActionItem.js:
```diff
@@ -198,12 +208,12 @@ class ReportActionItem extends Component {
                                         : (
                                             <ReportActionItemGrouped>
-                                                {children}
+                                                {this.renderItemContent(hovered)}
```
```suggestion
                                                {this.renderItemContent(hovered || this.state.isContextMenuActive)}
```

parasharrajat reviewed (changes requested):
Should fix https://github.com/Expensify/App/pull/12762#issuecomment-1324277609 ?

grgia committed:
Remain focused when context menu open

grgia committed:
Merge branch 'main' into georgia-IOU-branding

parasharrajat commented:
Thanks. Testing...

parasharrajat commented:
### Screenshots

#### :black_square_button: iOS / native
![Screenshot 2022-11-23 05:58:05](https://user-images.githubusercontent.com/24370807/203447622-1e6b6004-276c-4633-9920-4f3fbb0ea691.png)
![Screenshot 2022-11-23 05:57:54](https://user-images.githubusercontent.com/24370807/203447624-c81f4170-6b5b-4952-9750-359cd37333b1.png)
![Screenshot 2022-11-23 05:55:54](https://user-images.githubusercontent.com/24370807/203447626-5bf71502-8a9b-4992-adda-3ef34e56751a.png)

#### :black_square_button: iOS / Safari
![Screenshot 2022-11-23 05:23:24](https://user-images.githubusercontent.com/24370807/203444286-8596e80b-2381-496d-9f71-2d54469e8bf8.png)

![Screenshot 2022-11-23 05:20:46](https://user-images.githubusercontent.com/24370807/203444104-a730d4b5-4e80-44fb-a5ed-4425fd5af898.png)
![Screenshot 2022-11-23 05:20:37](https://user-images.githubusercontent.com/24370807/203444110-4cc3c593-26f5-46c0-b61e-7a33767b5092.png)
#### :black_square_button: MacOS / Desktop
![Screenshot 2022-11-23 05:17:45](https://user-images.githubusercontent.com/24370807/203443799-cf4fbc49-42d4-4526-89ed-7e82017a7633.png)
![Screenshot 2022-11-23 05:17:35](https://user-images.githubusercontent.com/24370807/203443807-7b22f7d6-da00-40be-9690-4e16f912b6a3.png)
![Screenshot 2022-11-23 05:09:19](https://user-images.githubusercontent.com/24370807/203443814-ac8b7938-4089-4238-8330-d18188566031.png)

#### :black_square_button: MacOS / Chrome

![Screenshot 2022-11-23 05:20:06](https://user-images.githubusercontent.com/24370807/203444140-00b954df-c6f9-4dcc-9772-307fa5f4ed16.png)
![Screenshot 2022-11-23 05:19:57](https://user-images.githubusercontent.com/24370807/203444144-c5cd7a08-73ec-40d8-9458-e32f0e8251eb.png)
![Screenshot 2022-11-23 05:19:19](https://user-images.githubusercontent.com/24370807/203444147-584808bc-69e7-44cf-9a4a-5d2a732214ad.png)

![Screenshot 2022-11-23 05:22:54](https://user-images.githubusercontent.com/24370807/203444239-1cf22d13-99b5-44ba-8a72-d8f5258ee7a5.png)

#### :black_square_button: Android / Chrome
![Screenshot 2022-11-23 06:05:54](https://user-images.githubusercontent.com/24370807/203448320-b0bd2ba1-1a78-43c5-bf5c-0f9eaa418329.png)
![Screenshot 2022-11-23 06:07:17](https://user-images.githubusercontent.com/24370807/203448464-544c2d43-eb8b-45f6-8640-754149ff54ed.png)

#### :black_square_button: Android / native

parasharrajat commented:
### Screenshots

#### :black_square_button: iOS / native
![Screenshot 2022-11-23 05:58:05](https://user-images.githubusercontent.com/24370807/203447622-1e6b6004-276c-4633-9920-4f3fbb0ea691.png)
![Screenshot 2022-11-23 05:57:54](https://user-images.githubusercontent.com/24370807/203447624-c81f4170-6b5b-4952-9750-359cd37333b1.png)
![Screenshot 2022-11-23 05:55:54](https://user-images.githubusercontent.com/24370807/203447626-5bf71502-8a9b-4992-adda-3ef34e56751a.png)

#### :black_square_button: iOS / Safari
![Screenshot 2022-11-23 05:23:24](https://user-images.githubusercontent.com/24370807/203444286-8596e80b-2381-496d-9f71-2d54469e8bf8.png)

![Screenshot 2022-11-23 05:20:46](https://user-images.githubusercontent.com/24370807/203444104-a730d4b5-4e80-44fb-a5ed-4425fd5af898.png)
![Screenshot 2022-11-23 05:20:37](https://user-images.githubusercontent.com/24370807/203444110-4cc3c593-26f5-46c0-b61e-7a33767b5092.png)
#### :black_square_button: MacOS / Desktop
![Screenshot 2022-11-23 05:17:45](https://user-images.githubusercontent.com/24370807/203443799-cf4fbc49-42d4-4526-89ed-7e82017a7633.png)
![Screenshot 2022-11-23 05:17:35](https://user-images.githubusercontent.com/24370807/203443807-7b22f7d6-da00-40be-9690-4e16f912b6a3.png)
![Screenshot 2022-11-23 05:09:19](https://user-images.githubusercontent.com/24370807/203443814-ac8b7938-4089-4238-8330-d18188566031.png)

#### :black_square_button: MacOS / Chrome

![Screenshot 2022-11-23 05:20:06](https://user-images.githubusercontent.com/24370807/203444140-00b954df-c6f9-4dcc-9772-307fa5f4ed16.png)
![Screenshot 2022-11-23 05:19:57](https://user-images.githubusercontent.com/24370807/203444144-c5cd7a08-73ec-40d8-9458-e32f0e8251eb.png)
![Screenshot 2022-11-23 05:19:19](https://user-images.githubusercontent.com/24370807/203444147-584808bc-69e7-44cf-9a4a-5d2a732214ad.png)

![Screenshot 2022-11-23 05:22:54](https://user-images.githubusercontent.com/24370807/203444239-1cf22d13-99b5-44ba-8a72-d8f5258ee7a5.png)

#### :black_square_button: Android / Chrome
![Screenshot 2022-11-23 06:05:54](https://user-images.githubusercontent.com/24370807/203448320-b0bd2ba1-1a78-43c5-bf5c-0f9eaa418329.png)
![Screenshot 2022-11-23 06:07:17](https://user-images.githubusercontent.com/24370807/203448464-544c2d43-eb8b-45f6-8640-754149ff54ed.png)

#### :black_square_button: Android / native

parasharrajat reviewed (dismissed):
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.


cc: @Luke9389 

:ribbon: :eyes: :ribbon:  C+ reviewed

grgia committed:
Merge branch 'main' into georgia-IOU-branding

grgia committed:
Fix IOU button

grgia commented:
<img width="507" alt="image" src="https://user-images.githubusercontent.com/38015950/203481077-bdfb9227-59b0-47c2-8340-2af2db781641.png">
Fixed with new button style

parasharrajat commented on src/styles/styles.js:
```diff
@@ -1657,7 +1657,7 @@ const styles = {
         borderWidth: 3,
         borderRadius: 18,
-        borderColor: themeColors.componentBG,
+        borderColor: themeColors.cardBG,
```
How is this change related to this PR?

grgia replied:
We want to avoid adding keys to the theme file in most situations. But in this case, I agree!

grgia committed:
Add cardBorder to themes.

parasharrajat reviewed (approved):
LGTM. 

cc: @shawnborton @Luke9389

parasharrajat commented on src/styles/styles.js:
```diff
@@ -374,9 +374,9 @@ const styles = {
         height: variables.componentSizeNormal,
-        paddingTop: 6,
+        paddingTop: 12,
         paddingRight: 16,
-        paddingBottom: 6,
+        paddingBottom: 12,
```
@shawnborton Can you please confirm the medium button vertical padding?

shawnborton replied:
I think this should be good. We could test quickly by having a letter in the button that has an descender, like "Again" so we can make sure that the bottom part of the g isn't cut off.

grgia replied:
on it! 
<img width="579" alt="image" src="https://user-images.githubusercontent.com/38015950/203608828-b5342794-0b04-49c3-9fe4-3b8119a7604a.png">

shawnborton replied:
Looks good!

shawnborton commented:
Looks good to me too. Luke is out today so feel free to merge @parasharrajat @MariaHCD

parasharrajat commented:
I can't merge this. cc: @MariaHCD

MariaHCD commented:
Thanks, all! I'm giving this a final review now.

MariaHCD commented:
@parasharrajat Re-ran the reviewer checklist check but it's still failing. Could you verify you've used the latest one? https://raw.githubusercontent.com/Expensify/App/main/contributingGuides/REVIEWER_CHECKLIST.md

parasharrajat commented:
I updated it thrice.

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/3536513612) to staging by @MariaHCD in version: 1.2.31-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

luacmartins commented:
The production deploy comment failed for this PR, but this was deployed to production on v1.2.32-2 on Nov 28.
